frame/system/src/extensions/check_genesis.rs

```diff
@@ -17,16 +17,17 @@
 
 use codec::{Encode, Decode};
 use crate::{Config, Pallet};
+use scale_info::TypeInfo;
 use sp_runtime::{
 	traits::{SignedExtension, Zero},
 	transaction_validity::TransactionValidityError,
 };
 
 /// Genesis hash check to provide replay protection between different networks.
-#[derive(Encode, Decode, Clone, Eq, PartialEq)]
-pub struct CheckGenesis<T: Config + Send + Sync>(sp_std::marker::PhantomData<T>);
+#[derive(Encode, Decode, Clone, Eq, PartialEq, TypeInfo)]
+pub struct CheckGenesis<T: Config + TypeInfo + Send + Sync>(sp_std::marker::PhantomData<T>);
```

Comment:
ideally the TypeInfo here should be removed. TypeInfo shouldn't require T implement TypeInfo

Reply:
Fixed with `skip_type_params`